support-utils/cre_astro_software_gather_logs.sh

```diff
@@ -0,0 +1,392 @@
+# echo "Gathering Node Status";kubectl get nodes -o wide > "$ASTRONOMER_NAMESPACE/Deployment_logs/nodes.log"
+# echo "Gathering kube-system pod status";kubectl get pods -o wide -n kube-system > "$ASTRONOMER_NAMESPACE/Deployment_logs/kube-system.log" 
+# echo "Gathering sevice Status in $ASTRONOMER_NAMESPACE Namespace ";kubectl get svc > "$ASTRONOMER_NAMESPACE/Deployment_logs/svc_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+# echo "Gathering persistent volume Status in $ASTRONOMER_NAMESPACE Namespace ";kubectl get pvc > "$ASTRONOMER_NAMESPACE/Deployment_logs/$PVC_DIR/pvc_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+# echo "Gathering ingress Status in $ASTRONOMER_NAMESPACE Namespace ";kubectl get ingress > "$ASTRONOMER_NAMESPACE/Deployment_logs/ingress_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+# echo "Gathering cronjobs Status in $ASTRONOMER_NAMESPACE Namespace ";kubectl get cronjobs > "$ASTRONOMER_NAMESPACE/Deployment_logs/cronjobs_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+# echo "Gathering jobs Status in $ASTRONOMER_NAMESPACE Namespace ";kubectl get jobs > "$ASTRONOMER_NAMESPACE/Deployment_logs/jobs_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+
+
+#Getting Some enviornment Info at root $DIR 
+echo "Gathering Diskspace of Registry pod in $ASTRONOMER_NAMESPACE Namespace ";kubectl get pods -n $ASTRONOMER_NAMESPACE  | grep registry | awk '{print $1}' | xargs -I {} kubectl exec -it -n $ASTRONOMER_NAMESPACE  {} -- df -Th > "$ASTRONOMER_NAMESPACE/Deployment_logs/Registry_Diskspace_$ASTRONOMER_NAMESPACE.log"
+echo "=======================Astro version output==========================================================================" > "$ASTRONOMER_NAMESPACE/Deployment_logs/Enviornment_Info.log"
+echo "Gathering Astro version status";astro version  >> "$DIR/Enviornment_Info.log"
+echo "=======================docker version output==========================================================================" >> "$ASTRONOMER_NAMESPACE/Deployment_logs/Enviornment_Info.log"
+echo "Gathering docker version status";docker version  >> "$DIR/Enviornment_Info.log"
+echo "=======================helm version output==========================================================================" >> "$ASTRONOMER_NAMESPACE/Deployment_logs/Enviornment_Info.log"
+echo "Gathering helm version status";helm version  >> "$DIR/Enviornment_Info.log"
+echo "Gathering helm status";helm ls -A >> "$DIR/helm_status.log"
+echo "======================Collecting Some General enviornment Information in the $ASTRONOMER_NAMESPACE======================"
+echo "Gathering get all status  in $ASTRONOMER_NAMESPACE Namespace";kubectl get all --all-namespaces > "$DIR/getall_status_$ASTRONOMER_NAMESPACE.log" -n $ASTRONOMER_NAMESPACE
+echo "Gathering All replica status in all namespaces";kubectl get rs --all-namespaces|grep -v '0         0         0' > "$DIR/rs_status_all_namespaces.log"
```

Comment:
Relying on the amount of whitespace in a grep is not the right way to do this. Consider using `awk '$1 == 0 && $2 == 0 && $3 == 0 {print}'` or something more precise than grep.